src/extended_range_proof.rs

```diff
@@ -212,6 +219,14 @@ where K: SecretKey
     pub minimum_value_promise: u64,
 }
 
+impl<K: SecretKey> Drop for ExtendedWitness<K>{
+    fn drop(&mut self) {
+        self.mask.zeroize();
+        self.value.zeroize();
+        self.minimum_value_promise.zeroize();
+    }
+}
```

Comment:
You should be able to get this functionality for free by deriving `ZeroizeOnDrop`.

Reply:
As a side note, `ZeroizeOnDrop` also has the benefit of being future-proof against structural changes.

Reply:
See [this commit](https://github.com/AaronFeickert/tari-crypto/commit/83ee9b7f95d1491d4221d81fb4be33852499ab91) for the derived traits.